spec/classes/broker_spec.rb

```diff
@@ -2,172 +2,140 @@
 require 'shared_examples_param_validation'
 
 describe 'kafka::broker', type: :class do
-  let :facts do
-    {
-      osfamily: 'Debian',
-      os: { family: 'Debian' },
-      operatingsystem: 'Ubuntu',
-      operatingsystemrelease: '14.04',
-      lsbdistcodename: 'trusty',
-      architecture: 'amd64',
-      service_provider: 'upstart'
-    }
-  end
-  let :common_params do
-    {
-      config: {
-        'zookeeper.connect' => 'localhost:2181'
-      }
-    }
-  end
-
-  let :params do
-    common_params
-  end
-
-  it { is_expected.to contain_class('kafka::broker::install').that_comes_before('Class[kafka::broker::config]') }
-  it { is_expected.to contain_class('kafka::broker::config').that_comes_before('Class[kafka::broker::service]') }
-  it { is_expected.to contain_class('kafka::broker::service').that_comes_before('Class[kafka::broker]') }
-  it { is_expected.to contain_class('kafka::broker') }
-
-  context 'on Debian' do
-    describe 'kafka::broker::install' do
-      context 'defaults' do
-        it { is_expected.to contain_class('kafka') }
+  on_supported_os.each do |os, facts|
```

Comment:
It's better to use `os_facts` as a variable name so you don't have a let block and a variable overlap.

Reply:
I just fixed it, after I took a closer look at the docs.